pkg/pquerier/frames.go

```diff
@@ -105,6 +108,7 @@ func NewDataFrame(columnsSpec []columnMeta, indexColumn Column, lset utils.Label
 		df.columns = make([]Column, 0, numOfColumns)
 		df.metricToCountColumn = map[string]Column{}
 		df.metrics = map[string]struct{}{}
+		df.nonEmptyRowsIndicator = make([]bool, columnSize)
```

Comment:
you can avoid this if you'll maintain simple int counter in a column for num of non empty rows

Reply:
`col.SetDataAt(..)` could increment/decrement the column counter